Firestore/core/src/model/value_util.cc

```diff
@@ -486,6 +491,105 @@ std::string CanonicalId(const google_firestore_v1_ArrayValue& value) {
   return CanonifyArray(value);
 }
 
+Message<google_firestore_v1_Value> GetLowerBound(pb_size_t value_tag) {
+  switch (value_tag) {
+    case google_firestore_v1_Value_null_value_tag:
+      return NullValue();
+
+    case google_firestore_v1_Value_boolean_value_tag: {
+      Message<google_firestore_v1_Value> value;
+      value->which_value_type = value_tag;
+      value->boolean_value = false;
+      return value;
+    }
+
+    case google_firestore_v1_Value_integer_value_tag:
+    case google_firestore_v1_Value_double_value_tag: {
+      return NaNValue();
+    }
+
+    case google_firestore_v1_Value_timestamp_value_tag: {
+      Message<google_firestore_v1_Value> value;
+      value->which_value_type = value_tag;
+      value->timestamp_value.seconds = std::numeric_limits<int64_t>::min();
+      value->timestamp_value.nanos = 0;
+      return value;
+    }
+
+    case google_firestore_v1_Value_string_value_tag: {
+      Message<google_firestore_v1_Value> value;
+      value->which_value_type = value_tag;
+      value->string_value = nullptr;
+      return value;
+    }
+
+    case google_firestore_v1_Value_bytes_value_tag: {
+      Message<google_firestore_v1_Value> value;
+      value->which_value_type = value_tag;
+      value->bytes_value = nullptr;
+      return value;
+    }
+
+    case google_firestore_v1_Value_reference_value_tag:
+      return RefValue(DatabaseId("", ""), DocumentKey::Empty());
```

Comment:
This access could cause a segfault since we are allocating a string. I think we can return a reference_value with a nullptr value, but would need to make sure the SDK treats this properly.